doc/api/fs.markdown

```diff
@@ -989,6 +989,32 @@ The synchronous version of [`fs.writeFile()`][]. Returns `undefined`.
 
 Synchronous versions of [`fs.write()`][]. Returns the number of bytes written.
 
+## fs.mkdtemp(template, callback)
+
+Creates a unique temporary directory.
+
+The `fs.mkdtemp()` function generates a uniquely named temporary
+directory from template. The last six characters of template must be
+XXXXXX and these are replaced with a string that makes the directory
+name unique.
+
+The created folder path is passed to the callback's second parameter.
+
+Example:
+
+```js
+fs.mkdtemp('/tmp/foo-XXXXXX', function(err, folder) {
+    console.log(folder); '/tmp/foo-itXde2'
```

Comment:
Usually the result of log statements will be expressed in comments I guess. cc @nodejs/documentation any convention we are following? like location of the result, comment, indentation of comment.